common/src/main/java/org/apache/celeborn/common/meta/FileManagedBuffers.java

```diff
@@ -56,6 +66,26 @@ public ManagedBuffer chunk(int chunkIndex, int offset, int len) {
     final long chunkLength = offsets[chunkIndex + 1] - chunkOffset;
     assert offset < chunkLength;
     long length = Math.min(chunkLength - offset, len);
+    MemCacheManager memCacheManager = MemCacheManager.getMemCacheManager();
+    chunkTotal++;
+    if (memCacheManager.contains(file.getAbsolutePath())) {
+      String filePath = file.getAbsolutePath();
+      ByteBuf fileBuffer = memCacheManager.getCache(filePath);
+      ByteBuf buffer = fileBuffer.slice((int) (chunkOffset + offset), (int) length);
```

Comment:
Seems we need to call `buffer.retain` here to support retried read. cc @RexXiong @FMX